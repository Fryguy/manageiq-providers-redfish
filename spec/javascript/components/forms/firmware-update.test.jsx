import FirmwareUpdate from '../../../../app/javascript/components/forms/firmware-update'

let renderComponent;

describe('FirmwareUpdate', () => {
  beforeAll(() => {
    renderComponent = (loading = false, physicalServers = [], firmwareBinaries = []) => shallowRedux(
      <FirmwareUpdate
        updateFormState={jest.fn()} loading={loading}
        physicalServerIds={physicalServers}
        firmwareBinaries={firmwareBinaries}
      />
      );
  });

  describe('renders', () => {
    it('spinner', () => {
      let component = renderComponent(true);
      expect(toJson(component)).toMatchSnapshot();
    });

    describe('form', () => {
      it('with zero physicalServers', () => {
        let component = renderComponent(false);
        expect(toJson(component)).toMatchSnapshot();
      });

      it('with two physicalServers', () => {
        let component = renderComponent(false, ['server1', 'server2']);
        expect(toJson(component)).toMatchSnapshot();
      });

      it('with two physicalServers and one firmware binary', () => {
        let component = renderComponent(false, ['server1', 'server2'], [{value: 'binary1', label: 'BINARY1'}]);
        expect(toJson(component)).toMatchSnapshot();
      });
    });
  });
});
