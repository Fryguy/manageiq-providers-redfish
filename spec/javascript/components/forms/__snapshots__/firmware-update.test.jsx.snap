// Jest Snapshot v1, https://goo.gl/fbAQLP

exports[`FirmwareUpdate renders form with two physicalServers 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormSpy
    onChange={[Function]}
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <h2>
          _Number of servers to update firmware for: 2_
        </h2>
      </Col>
    </Row>
    <hr />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          inputColumnSize={8}
          label="_Firmware Binary_"
          labelColumnSize={3}
          name="firmwareBinary"
          options={Array []}
          parse={[Function]}
          placeholder="_Select a Firmware Binary_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <hr />
    </Row>
  </Grid>
</Form>
`;

exports[`FirmwareUpdate renders form with two physicalServers and one firmware binary 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormSpy
    onChange={[Function]}
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <h2>
          _Number of servers to update firmware for: 2_
        </h2>
      </Col>
    </Row>
    <hr />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          inputColumnSize={8}
          label="_Firmware Binary_"
          labelColumnSize={3}
          name="firmwareBinary"
          options={
            Array [
              Object {
                "label": "BINARY1",
                "value": "binary1",
              },
            ]
          }
          parse={[Function]}
          placeholder="_Select a Firmware Binary_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <hr />
    </Row>
  </Grid>
</Form>
`;

exports[`FirmwareUpdate renders form with zero physicalServers 1`] = `
<Form
  bsClass="form"
  componentClass="form"
  horizontal={true}
  inline={false}
>
  <FormSpy
    onChange={[Function]}
  />
  <Grid
    bsClass="container"
    componentClass="div"
    fluid={true}
  >
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <h2>
          _Number of servers to update firmware for: 0_
        </h2>
      </Col>
    </Row>
    <hr />
    <Row
      bsClass="row"
      componentClass="div"
    >
      <Col
        bsClass="col"
        componentClass="div"
        xs={12}
      >
        <Field
          component={[Function]}
          format={[Function]}
          inputColumnSize={8}
          label="_Firmware Binary_"
          labelColumnSize={3}
          name="firmwareBinary"
          options={Array []}
          parse={[Function]}
          placeholder="_Select a Firmware Binary_"
          searchable={true}
          validate={[Function]}
          validateOnMount={false}
        />
      </Col>
      <hr />
    </Row>
  </Grid>
</Form>
`;

exports[`FirmwareUpdate renders spinner 1`] = `
<div
  className="spinner spinner-lg"
/>
`;
